app_server/app/adapters/socket/Room.js

```diff
@@ -75,4 +75,21 @@ Class.removeAllClients = function() {
   this.__clients = {};
 };
 
+Class.getNumberOfUsers = function() {
+  let users = {};
+  let count = 0;
+  for (var socketId in this.getClients()) {
+    let userId = this.getClient(socketId).getUserId(); 
+    if (!users[userId]) {
+      users[userId] = true;
+      count++;
+    }
+  }
+  return count;
```

Comment:
Yeah, that should work, `map` then `reduce` to a js object. But I suppose that will then make this a bit more complex than the current procedure flow and might not worth it. Also quite troublesome to test this part properly again (need to run wowza locally and try streaming/commenting in different scenarios) so I will leave this for now.